cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,485 @@
+            for topic, topic_words in top_n_words.items():
+                words = [word[0] for word in topic_words]
+                word_embeddings = self.create_embeddings(words)
+                topic_embedding = self.create_embeddings(
+                    " ".join(words)
+                ).reshape(1, -1)
+                topic_words = mmr(
+                    topic_embedding,
+                    word_embeddings,
+                    words,
+                    top_n=n,
+                    diversity=0
+                )
+                top_n_words[topic] = [
+                    (word, value)
+                    for word, value in top_n_words[topic]
+                    if word in topic_words
+                ]
+
+        top_n_words = {
+            label: values[:n] for label, values in top_n_words.items()
+        }
+
+        topic_names = {
+            key: f"{key}_" + "_".join([word[0] for word in values[:4]])
+            for key, values in top_n_words.items()
+        }
+        return top_n_words, topic_names
+
+    def extract_topic_sizes(self, df):
+        """Calculate the topic sizes
+        Arguments:
+            documents: dataframe with documents and their corresponding IDs
+            and added Topics
+
+        Returns:
+            topic_sizes: DataFrame containing topic cluster sizes
+        """
+        topic_sizes = (
+            df.groupby(["Topic"])
+            .Document.count()
+            .reset_index()
+            .rename({"Topic": "Topic", "Document": "Count"}, axis="columns")
+            .sort_values("Count", ascending=False)
+        )
+        return topic_sizes
+
+    # Topic reduction
+    def reduce_topics(self, num_topics, tf_idf, docs_df):
+        """Reduce topics to num_topics
+        Arguments:
+            tf_idf: c_tf_idf matrix obtained
+            docs_df: Dataframe with documents and their corresponding IDs
+            and Topics
+        Returns:
+            docs_df: Updated dataframe with documents and the reduced
+            number of Topics
+            top_n_words: top n words based on the updated dataFrame
+        """
+        for i in range(num_topics):
+            # Calculate cosine similarity
+            similarities = pairwise_distances(tf_idf, metric="cosine")
+            cp.fill_diagonal(similarities, 0)
+
+            # Extract label to merge into and from where
+            topic_sizes = (
+                docs_df.groupby(["Topic"])
+                .count()
+                .sort_values("Document", ascending=False)
+                .reset_index()
+            )
+            topic_to_merge = topic_sizes.iloc[-1]["Topic"]
+            topic_to_merge_into = cp.argmax(
+                similarities[topic_to_merge + 1]) - 1
+
+            # Adjust topics
+            a = cudf.Series(topic_to_merge_into)
+            docs_df.loc[docs_df["Topic"] == topic_to_merge, "Topic"] = a
+            old_topics = docs_df.sort_values("Topic").Topic.unique()
```

Comment:
Why are you sorrting here if all you need is unique values . Do you need sorted unique values ? 

In that case i suggest doing something like below. 
```suggestion
            old_topics = docs_df.Topic.unique().sort_values()
```